File: stackoverflow_example.py
# -*- coding: utf-8 -*-
import pywt
from wrcoef import wavedec, wrcoef


# Question 1

# Generate input data
DATA = range(10)

# Define wavelet parameters
N_LEVELS = 2
WAVELET_NAME = 'sym4'

# Define wavelet
w = pywt.Wavelet(WAVELET_NAME)

# Decompose input signal
C, L = wavedec(DATA, wavelet=w, level=N_LEVELS)

# Reconstruct all the coefficients
for n in range(len(L)-2):
    D = wrcoef(C, L, wavelet=w, level=n+1)
    print(n+1, D)


# Question 2

# Generate input data
DATA = range(18)

# Define wavelet parameters
N_LEVELS = 4
WAVELET_NAME = 'sym2'

# Define wavelet
w = pywt.Wavelet(WAVELET_NAME)

# Decompose input signal
C, L = wavedec(DATA, wavelet=w, level=N_LEVELS)

# Reconstruct all the coefficients
for n in range(len(L)-2):
    D = wrcoef(C, L, wavelet=w, level=n+1)
    print(n+1, D)

File: stackoverflow_example.m
%% Stackoverflow questions
% https://stackoverflow.com/questions/45051106/multilevel-partial-wavelet-reconstruction-with-pywavelets/47591523
% and
% https://dsp.stackexchange.com/questions/42358/multilevel-partial-wavelet-reconstruction-with-pywavelets/42372
clc; clear all; format short

%% Question 1
DATA = 0:9;
N_LEVELS = 2;
WAVELET_NAME = 'sym4';
[C, L] = wavedec(DATA, N_LEVELS, WAVELET_NAME);
for n=1:max(1, length(L)-2), disp(n)
    D = wrcoef('d', C, L, WAVELET_NAME, n);
    disp(D)
end


%% Question 2
DATA = 0:17;
N_LEVELS = 4;
WAVELET_NAME = 'sym2';
[C, L] = wavedec(DATA, N_LEVELS, WAVELET_NAME); 
for n=1:max(1, length(L)-2), disp(n)
    D = wrcoef('d', C, L, WAVELET_NAME, n);
    disp(D)
end